src/classes/CRLP_RecalculateBTN_TEST.cls

```diff
@@ -172,4 +172,367 @@ private class CRLP_RecalculateBTN_TEST {
+        );
+        insert new List<Account>{household, organization};
+
+        Contact contact = new Contact(
+            LastName = 'TEST_CONTACT',
+            AccountId = household.Id
+        );
+        insert contact;
+
+        // Create the Contact Household and the Organization donation
+        Opportunity householdContactDonation = new Opportunity(
+            Name = 'TEST_CONTACT_DONATION',
+            StageName = UTIL_UnitTestData_TEST.getClosedWonStage(),
+            CloseDate = Date.today(),
+            Amount = 100,
+            AccountId = household.Id,
+            Primary_Contact__c = contact.Id
+        );
+        Opportunity organizationDonation = new Opportunity(
+            Name = 'TEST_ORGANIZATION_DONATION',
+            StageName = UTIL_UnitTestData_TEST.getClosedWonStage(),
+            CloseDate = Date.today(),
+            Amount = 100,
+            AccountId = organization.Id,
+            Primary_Contact__c = contact.Id
+        );
+        insert new List<Opportunity>{householdContactDonation, organizationDonation};
+
+        // Make the Contact as Primary Contact on both donations.
+        List<OpportunityContactRole> contactOCRs = new List<OpportunityContactRole>{
+            new OpportunityContactRole(
+                OpportunityId = householdContactDonation.Id,
+                ContactId = contact.Id,
+                IsPrimary = true,
+                Role = role
+            ),
+            new OpportunityContactRole(
+                OpportunityId = organizationDonation.Id,
+                ContactId = contact.Id,
+                IsPrimary = false,
+                Role = role
+            )
+        };
+        insert contactOCRs;
+
+        Partial_Soft_Credit__c contactPartialSoftCredit = new Partial_Soft_Credit__c(
+            Amount__c = 75,
+            Contact_Role_ID__c = contactOCRs[1].Id,
+            Contact__c = contactOCRs[1].ContactId,
+            Opportunity__c = contactOCRs[1].OpportunityId,
+            Role_Name__c = contactOCRs[1].Role
+        );
+        insert contactPartialSoftCredit;
+
+        activateTriggers();
+
+        Test.startTest();
+        CRLP_RecalculateBTN_CTRL controller = new CRLP_RecalculateBTN_CTRL(new ApexPages.StandardController(contact));
+        controller.buttonClick();
+        Test.stopTest();
+
+        List<Contact> updatedContact = [
+            SELECT npo02__Soft_Credit_Total__c
+            FROM Contact
+            WHERE Id = :contact.Id
+        ];
+        System.assertEquals(contactPartialSoftCredit.Amount__c, updatedContact[0].npo02__Soft_Credit_Total__c);
+    }
+
+    /*******************************************************************************************************
+    * @description Deactivates all the Triggers.
+    * @return void.
+    ********************************************************************************************************/
+    private static void deactivateTriggers() {
```

Comment:
I'm curious why the tests need to deactivate and then reactivate all the triggers? Also there are methods in `UTIL_UnitTestData_TEST` that will disable specific triggers by type.

Reply:
The Triggers are deactivated during the creation of the test data to ensure that the Triggers don't alter the test data. After the test data are created, the Triggers are activated.

Reply:
Do you need to deactivate all of the triggers, or just specific ones?